src/main/scala/com/dwolla/sbt/s3/PublishToS3.scala

```diff
@@ -28,12 +28,14 @@ object PublishToS3 extends AutoPlugin {
   )
 
   lazy val tasks = Seq(
-    s3PublishSnapshot := !(git.gitCurrentBranch.value == "master" && !git.gitUncommittedChanges.value),
+    s3PublishSnapshot := !(
+      (git.gitCurrentBranch.value == "master" || git.gitHeadCommit.value.contains(git.runner.value("rev-parse master")(baseDirectory.value, streams.value.log)))
```

Comment:
This bit might be more clear if we extracted out some functions, e.g. one named `masterBranchIsCheckedOut` for the new predicate. That's a little tricky because of how sbt uses macros to implement `.value`, so I think it would look something like 

```scala
    s3PublishSnapshot := !(
      !git.gitUncommittedChanges.value && (
        git.gitCurrentBranch.value == "master" ||
          masterBranchIsCheckedOut("master", git.gitHeadCommit.value, git.runner.value, baseDirectory.value, streams.value.log)
        )
      ),
```

and then

```scala
  private def masterBranchIsCheckedOut(branch: String, headCommit: Option[String], gitRunner: GitRunner, workingDirectory: File, log: Logger): Boolean =
    headCommit.contains(gitRunner(s"rev-parse $branch")(workingDirectory, log))
```

(`.value` is a macro that helps implement sbt's dependency graph and effect management, and it can only be called within a task or setting macro. In this case that means we need to keep the `.value`s on the right side of the `:=` above, and pass the values they return to our extracted function.)

Reply:
Renamed to `publishBranchIsCheckedOut` due to the other comment about removing the hard coded-ness of `master`